boot/bootchain.go

```diff
@@ -270,7 +270,20 @@ func bootAssetsToLoadChains(assets []bootAsset, kernelBootFile bootloader.BootFi
 		return nil, fmt.Errorf("internal error: no bootloader name for boot asset role %q", thisAsset.Role)
 	}
 	var chains []*secboot.LoadChain
-	for _, hash := range thisAsset.Hashes {
+
+	for i, hash := range thisAsset.Hashes {
+		// There should be 1 or 2 assets, and their position has a meaning.
+		// See TrustedAssetsUpdateObserver.observeUpdate
+		if i == 0 {
```

Comment:
It looks like i==0 is the old asset and i==1 the new one, but please add a comment to clarify, also worth a longer comment saying basically what you say in the commit, I think it is important enough so we keep that in mind while reading the code.